runtime/common/compositekind.go

```diff
@@ -180,3 +182,20 @@ func (k CompositeKind) SupportsInterfaces() bool {
 func (k CompositeKind) MarshalJSON() ([]byte, error) {
 	return json.Marshal(k.String())
 }
+
+var compositeKindMap = map[string]CompositeKind{
+	"CompositeKindUnknown":   CompositeKindUnknown,
+	"CompositeKindStructure": CompositeKindStructure,
+	"CompositeKindResource":  CompositeKindResource,
+	"CompositeKindContract":  CompositeKindContract,
+	"CompositeKindEvent":     CompositeKindEvent,
+	"CompositeKindEnum":      CompositeKindEnum,
+}
+
+func (a *CompositeKind) UnmarshalJSON(b []byte) error {
+	if c, ok := compositeKindMap[strings.Trim(string(b), `"`)]; ok {
```

Comment:
Instead of trimming the serialized string's quotes manually, potentially leaving quoted content, unmarshal as a string:

```go
var s string
if err := json.Unmarshal(b, &s); err != nil {
	return err
}
```